src/stream_type.rs

```diff
@@ -52,4 +54,38 @@ impl io::Write for StreamType {
     }
 }
 
-impl Io for StreamType {}
+impl AsyncRead for StreamType {
+    unsafe fn prepare_uninitialized_buffer(&self, buf: &mut [u8]) -> bool {
```

Comment:
In general no sane implementation of `Read` ever needs to read from the buffer it's writing into, but since safe rust can't assume that, this fn must be unsafe.